codecov_cli/parsers/base.py

```diff
@@ -0,0 +1,60 @@
+from datetime import datetime, timedelta
+from typing import List
+
+
+class ParsingError(Exception):
+    ...
+
+
+class Test(object):
+    def __init__(self, name: str, status: bool, duration: timedelta):
+        self.name = name
+        self.status = status
+        self.duration = duration
+
+    def __repr__(self):
+        return f"{self.name}:{self.status}:{self.duration.total_seconds()}::"
+
+
+class Testrun(object):
+    def __init__(
+        self,
+        name: str,
+        timestamp: datetime,
+        time: timedelta,
+        testcases: List[Test],
+        failures: int,
+        errors: int,
+        skipped: int,
+        total: int,
+    ):
+        self.testcases = testcases or []
+        self.name = name
+        self.time = time
+        self.timestamp = timestamp
+        self.failures = failures
+        self.errors = errors
+        self.skipped = skipped
+        self.total = total
+
+    def to_str(self):
```

Comment:
The uploader uploads JSON data... maybe we should have the encoding/decoding be to JSON?

If this is for debug purposes (e.g. printing the class nicely into console) than I believe the right function to override is `__repr__`.